training/CHAPEL_TUTORIAL.md

```diff
@@ -482,6 +531,109 @@ Promotion is equivalent to a `forall` loop. For this example the equivalent loop
 [x in [1, 2, 3, 4, 5]] factorial(x);
 ```
 
+<a id="filter"></a>
+## Filtering
+<a id="ex15"></a>
+#### Example 15: Filtering
+We can filter an iterator to only operate on values matching a certain condition:
+```chapel
+writeln([i in 0..#10] if i%2 == 0 then -i);
+```
+```console
+0 -2 -4 -6 -8
+```
+
+<a id="TIY3"></a>
+#### Try It Yourself 3: Sum Odd Perfect Squares <=25
+Problem:
+Use filtering and reduce to sum all odd perfect squares less than or equal to `25`
+<details>
+  <summary>Potential Solution</summary>
+
+```Chapel
+writeln(+ reduce [i in 0..5] if i%2 != 0 then i**2);
+```
+</details>
+Expected Output:
+
+```console
+35
+```
+
+<a id="locality"></a>
+## Locality
+<a id="locale"></a>
+### Locales
+[Locales](https://chapel-lang.org/docs/users-guide/locality/localesInChapel.html)
+can be thought of as chunk of memory that can do computation. Things that are co-located within a single locale
+are close to each other in the system and can interact with one another relatively cheaply. Things that are in distinct
+locales can still interact with each other in the same ways, but this is more expensive.
```

Comment:
I would add mention that the expense is due to the locales having to communicate/transfer data. Otherwise, if someone is not familiar with distributed computing this will seem extremely vague.

Reply:
yah this is a bit handwavy, ill update this